### PR TITLE
Add SEClamp support in CoreNEURON

### PR DESCRIPTION
**Pull request by pramodk**
- add svclmp.mod mod file from NEURON
 - update registeration callbacks
 - expose at_time in the header file

### CODE REVIEW

**Comment by pramodk** [comment]
NMODL needs to handle `after_cvode`. Also, `at_time` function needs some extra annotation for GPU:

```

PGCC/x86-64 Linux 19.10-0: compilation completed with warnings
PGCC-S-1000-Call in OpenACC region to procedure '_ZN10coreneuron7at_timeEPNS_9NrnThreadEd' which has no acc routine information (/jenkins/07/workspace/oss.coreneuron.pipeline/build_GPU-non-unified/coreneuron/svclmp.cpp: 434)
PGCC-W-0155-Compiler failed to translate accelerator region (see -Minfo messages): Missing branch target block (/jenkins/07/workspace/oss.coreneuron.pipeline/build_GPU-non-unified/coreneuron/svclmp.cpp: 336)
PGCC-F-0704-Compilation aborted due to previous errors. (/jenkins/07/workspace/oss.coreneuron.pipeline/build_GPU-non-unified/coreneuron/svclmp.cpp)
PGCC/x86-64 Linux 19.10-0: compilation aborted

```

**Comment by cattabiani** [comment]
please retest

**Comment by cattabiani** [comment]
Given the recent work done by @iomaganaris on at_time, I think this PR can be merged too right?

**Comment by pramodk** [comment]
Yes but nmodl submodule needs to be updated in coreneuron repo.

**Comment by pramodk** [comment]
```
Warning: No --target specified on command-line. Using default system target 
3321        "avx512skx-i32x16". 
3322Warning: No --target specified on command-line. Using default system target 
3323        "avx512skx-i32x16". 
3324x86_64/core/mod2c/svclmp.ispc:83:13: Error: Undeclared symbol "at_time". 
3325            at_time(nt, inst->dur1[id]);
3326            ^^^^^^^
3327
3328x86_64/core/mod2c/svclmp.ispc:86:13: Error: Undeclared symbol "at_time". 
3329            at_time(nt, inst->tc2[id]);
3330            ^^^^^^^
3331
3332x86_64/core/mod2c/svclmp.ispc:89:13: Error: Undeclared symbol "at_time". 
3333            at_time(nt, inst->tc3[id]);
3334            ^^^^^^^
```
I don't remember how to expose "annotate" function for ISPC but @ohm314 or @iomaganaris  might remember better.

File: coreneuron/mechanism/nrnoc_ml.ispc [hunk 1]
```diff
@@ -166,4 +166,7 @@ struct NrnThread {
     void* mapping;
 };
 
-extern uniform double ispc_celsius;
\ No newline at end of file
+extern uniform double ispc_celsius;
+
+extern "C"
+unmasked int at_time(NrnThread* nt, varying double te);
```
Comment (pramodk) [inline coreneuron/mechanism/nrnoc_ml.ispc:172]: as discussed offline, considering semantics of `unmasked`, we could have something like:
```
if (something)
   at_time(...)
```

And hence we can remove unmasked from here.

**Review by iomaganaris — APPROVED** [review]

**Comment by pramodk** [comment]
Thanks @iomaganaris for all fixes with ISPC!

**Pull request merged**